### PR TITLE
Show empty library in case search doesn't match anything

### PR DESCRIPTION
**Pull request by schroda**
In case the search didn't match anything all mangas where shown instead of an empty library.

### CODE REVIEW

File: src/components/library/LibraryMangaGrid.tsx [hunk 1]
```diff
@@ -48,30 +48,25 @@ const queryGenreFilter = (query: NullAndUndefined<string>, { genre }: IMangaCard
...
-        }, {});
-        filteredManga = Object.values(unique);
-    }
-    filteredManga = (filteredManga.length ? filteredManga : manga).filter(
-        (m) => downloadedFilter(downloaded, m) && unreadFilter(unread, m),
-    );
-    return filteredManga;
-};
+): IMangaCard[] =>
+    performSearch(query, manga).filter((m) => downloadedFilter(downloaded, m) && unreadFilter(unread, m));
```
Comment (akabhirav) [inline src/components/library/LibraryMangaGrid.tsx:69]: filtering the manga, before calling performSearch should increase the performance for large libraries

Comment (akabhirav) [inline src/components/library/LibraryMangaGrid.tsx:69]:
```suggestion
    performSearch(query, manga.filter((m) => downloadedFilter(downloaded, m) && unreadFilter(unread, m)));
```
something like this

Comment (schroda) [reply src/components/library/LibraryMangaGrid.tsx:69]: you're right it's not optimal currently.
I think the best thing should be to perform search and filter on each manga in one go.
I'll open a new pr for that

Comment (schroda) [reply src/components/library/LibraryMangaGrid.tsx]: nervermind, changed it in this pr

**Review by akabhirav — APPROVED** [review]
LGTM

**Pull request merged**